### PR TITLE
validation: once `Fail`, never go back to `Ignore`

### PR DESCRIPTION
**Pull request by howardjohn**
Currently, when an Istiod becomes leader it will check if its "ready" to
serve validation, and if so patch it to `Fail` from `Ignore`, so it
fails-closed.

This is intended for the initial install case; we don't want to block
all config initially. This is less important now that
https://github.com/istio/istio/issues/22095 is done, but some folks
still deploy their whole system at once so its probably nice to keep
around.

However, what we are also doing is transition from Fail -> Ignore. IMO
this is unsafe.

This PR makes it so its a one way transition from Ignore -> Fail.

### CODE REVIEW

**Comment by howardjohn** [comment]
Put a hold for more discussion in case its controversial. Ill add some tests if we want to go down this route

**Comment by howardjohn** [comment]
@ramaraochavali @hzxuzhonghu wdyt?

**Comment by hzxuzhonghu** [comment]
one way transition looks good, transiting from failure to Ignore could make istiod accept and handle invalid CRs, could break down it, with crash

**Comment by ramaraochavali** [comment]
+1 for one way transition

File: pkg/webhooks/validation/controller/controller.go [hunk 1]
```diff
@@ -263,37 +260,42 @@ func (c *Controller) isDryRunOfInvalidConfigRejected() (rejected bool, reason st
...
+		if ready {
+			updated.Webhooks[i].FailurePolicy = ptr.Of(kubeApiAdmission.Fail)
+		}
 	}
 
 	latest, err := c.webhooks.Update(updated)
 	if err != nil {
-		scope.Errorf("Failed to update validatingwebhookconfiguration %v (failurePolicy=%v, resourceVersion=%v): %v",
-			updated.Name, failurePolicy, updated.ResourceVersion, err)
+		scope.Errorf("failed to updated: %v", err)
```
Comment (stevenctl) [inline pkg/webhooks/validation/controller/controller.go:293]: Super nit:

```suggestion
		scope.Errorf("failed to update: %v", err)
```

**Review by stevenctl — APPROVED** [review]

**Pull request merged**